### PR TITLE
Fixed static analysis warnings in SimG4Core

### PR DESCRIPTION
**Pull request by civanch**
#### PR description:
Attempt to reduce number of warnings from static analyzer. Regression is expected.

#### PR validation:
Private

#### if this PR is a backport please specify the original PR and why you need to backport that PR:
no

### CODE REVIEW

**Comment by civanch** [comment]
please test

**Comment by civanch** [comment]
+1

File: SimG4Core/Application/src/RunManagerMTWorker.cc [hunk 1]
```diff
@@ -255,17 +255,20 @@ void RunManagerMTWorker::initializeG4(RunManagerMT* runManagerMaster, const edm:
...
+                                         thisID);
   } else if (uitype == "FilePerThread") {
-    new CustomUIsessionToFile(m_pCustomUIsession.getUntrackedParameter<std::string>("ThreadFile", ""), thisID);
+    ui = new CustomUIsessionToFile(m_pCustomUIsession.getUntrackedParameter<std::string>("ThreadFile", ""), thisID);
   } else {
     throw edm::Exception(edm::errors::Configuration)
         << "RunManagerMTWorker::initializeG4: Invalid value of CustomUIsession.Type '" << uitype
         << "', valid are MessageLogger, MessageLoggerThreadPrefix, FilePerThread";
   }
+  G4UImanager::GetUIpointer()->SetCoutDestination(ui);
```
Comment (perrotta) [inline SimG4Core/Application/src/RunManagerMTWorker.cc:271]: Where is this `ui` deleted?

**Comment by perrotta** [comment]
@civanch , since you are fixing the issues reported by the static analyzer, maybe you can also take care of
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-9b690f/20905/llvm-analysis/report-c64a1b.html#EndPath
(perhaps it is enough throwing at L89 also if `regionName` is still equal to `""`)

File: SimG4Core/Application/src/RunManagerMTWorker.cc (hunk 1 above)
Comment (civanch) [reply SimG4Core/Application/src/RunManagerMTWorker.cc:271]: Normally it is Geant4 responsibility - G4UImanager should do that but in Geant4 10.7 it is not the case... Will address.

**Comment by civanch** [comment]
please test

**Comment by civanch** [comment]
+1

File: SimG4Core/Application/src/RunManagerMTWorker.cc (hunk 1 above)
Comment (civanch) [reply SimG4Core/Application/src/RunManagerMTWorker.cc]: Is fixed.

**Comment by perrotta** [comment]
+1

**Pull request merged**